src/librbd/image/CreateRequest.cc

```diff
@@ -276,80 +277,19 @@ void CreateRequest<I>::validate_data_pool() {
 
   ldout(m_cct, 15) << dendl;
 
-  using klass = CreateRequest<I>;
-  librados::AioCompletion *comp =
-    create_rados_callback<klass, &klass::handle_validate_data_pool>(this);
-
-  librados::ObjectReadOperation op;
-  op.read(0, 0, nullptr, nullptr);
-
-  m_outbl.clear();
-  int r = m_data_io_ctx.aio_operate(RBD_INFO, comp, &op, &m_outbl);
-  ceph_assert(r == 0);
-  comp->release();
+  auto ctx = create_context_callback<
+    CreateRequest<I>, &CreateRequest<I>::handle_validate_data_pool>(this);
+  auto req = ValidatePoolRequest<I>::create(m_data_io_ctx, m_op_work_queue,
+                                            ctx);
+  req->send();
```

Comment:
Just thinking. It looks like we are doing this validation for every namespace. Don't we want to improve it and validate it for no namespace `m_data_io_ctx`? Or probably `ValidatePoolRequest` could dup the `io_cxt` and reset its namespace.

Reply:
:+1: